azext_iot/operations/hub.py

```diff
@@ -148,6 +154,20 @@ def iot_device_list(
     return result
 
 
+def _iot_device_list(hub_name, target, edge_enabled=False, top=1000):
```

Comment:
target should have hub name already - fix this and the other helper functions you added

Reply:
target does not have the hub name directly, but it does have entity name, which includes hub name. for my hub, entity name is "ms-hub.azure-devices.net". Is entity name guaranteed to always be of the form "hubname.azure-devices.net"?

Reply:
oh, seems like it is

Reply:
you are guaranteed that the stuff before the first . is the hub name - but the hostname (azure-devices.net) is not guaranteed for preview environments and features